app/src/main/java/com/github/orkest/View/sharing/SharingActivity.kt

```diff
@@ -63,35 +88,124 @@ class SharingComposeActivity : ComponentActivity() {
         }
     }
 
+    /**
+     * This function is used to connect to the Spotify API and get the song name
+     *
+     */
+    private fun spotifyAuthorization(){
+        // This is a callback for the authorization function that requests the access token
+        val showLoginActivityToken = registerForActivityResult(
+            ActivityResultContracts.StartActivityForResult()
+        ) { result: ActivityResult ->
+
+            val authorizationResponse = AuthorizationClient.getResponse(result.resultCode, result.data)
+
+            // Check the response type
+            when (authorizationResponse.type) {
+                AuthorizationResponse.Type.TOKEN -> {
+                    // Retrieve the access token
+                    this.accessToken = authorizationResponse.accessToken
+                    Log.d("SPOTIFY TOKEN", this.accessToken)
+                    CompletableFuture.runAsync {
+                        getTrackName(this.spotifySongID)
+                    }
+                }
+                AuthorizationResponse.Type.ERROR ->
+                    Log.d("SPOTIFY ERROR", authorizationResponse.error)
+                else ->
+                    Log.d("SPOTIFY FATAL ERROR", "Abort")
+            }
+        }
+
+        // This is a callback for the authorization function that requests the authorization code
```

Comment:
Maybe you could add it in a separate method to modularize your code and improve readability